### PR TITLE
ch configure dotenv #14

### PR DESCRIPTION
**Pull request by 13XAVI**
#14 Environmental Configuration

#### What does this PR do?
- This PR configure all variables that are being used in the  project and hide all sensitive information  including links,ports and  others so that  they won't be  pushed 
#### Description of Task to be completed?
- The task involves setting up a .env file to store environmental variables and sensitive information such as API keys, database credentials, and other configuration settings.
- Ensure that all necessary environmental variables are defined in the .env file and are appropriately referenced throughout the project.
- Implement logic within the project to read these environmental variables from the .env file during runtime.
Ensure that any sensitive information is securely managed and not exposed in the codebase or version control system.
- Hide all sensitive information
### Manually test by:
- Creating a .env file in the project root directory.
- Adding necessary environmental variables with appropriate values (e.g., API_KEY, DB_CONNECTION_STRING).
- Running the project and verifying that it reads the environmental variables from the .env file correctly.
- Ensuring that sensitive information is not hard-coded in the codebase but instead accessed through environmental variables.

### CODE REVIEW

**Review by wayneleon1 — APPROVED** [review]
Looks good

File: package.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
     "name": "dynamites-ecomm-be",
     "version": "1.0.0",
     "description": "",
-    "main": "index.js",
+    "main": "src/index.ts",
     "scripts": {
-        "start": "node ./dist/index.js",
+        "start": "nodemon --watch",
```
Comment (Habinezajanvier) [inline package.json]: Can we stick to node for production use

File: src/database/index.ts [hunk 2]
```diff
@@ -1,13 +1,14 @@
 import { DataSource } from 'typeorm';
 import { config } from 'dotenv';
 config();
+const dabasePort = process.env.PORT ? parseInt(process.env.PORT) :undefined
```
Comment (Habinezajanvier) [inline src/database/index.ts]: Let's stick to DB_PORT rather than PORT to alleviate future confusions

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (TresorXavier) [reply package.json]: i think this directory and  file does not exist /dist/index.js

File: dist/controller/user.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    user.firstName = data.firstName;
+    user.lastName = data.lastName;
+    user.age = data.age;
+    return await userRepository.save(user);
+};
+exports.createUser = createUser;
+const allUsers = async () => {
+    return await userRepository.find();
+};
+exports.allUsers = allUsers;
```
Comment (Habinezajanvier) [inline dist/controller/user.js]: Please put `dist/` in your `.gitignore`

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

**Review by Habinezajanvier — APPROVED** [review]
👍

**Pull request merged**